buttercup/cogs/find.py

```diff
@@ -1,69 +1,238 @@
+
+    def to_embed(self) -> Embed:  # noqa: C901
+        """Convert the submission to a Discord embed."""
+        color = Color.from_rgb(255, 176, 0)  # Orange
+        status: str = "Unclaimed"
+
+        tr_link = f"[Link]({self.transcription['url']})" if self.transcription else None
+
+        if self.submission["completed_by"]:
+            color = Color.from_rgb(148, 224, 68)  # Green
+
+            if self.author:
+                status = "Completed by {}".format(
+                    DISCORD_USERNAME_LINK.format(self.author["username"])
+                )
+            else:
+                status = "Completed"
+        elif self.submission["claimed_by"]:
+            color = Color.from_rgb(13, 211, 187)  # Cyan
+
+            if self.author:
+                status = "Claimed by {}".format(
+                    DISCORD_USERNAME_LINK.format(self.author["username"])
+                )
+            else:
+                status = "Claimed"
+
+        embed = (
+            Embed(color=color)
+            .add_field(name="Status", value=status)
+            .add_field(
+                name="OCR",
+                value="Yes" if self.submission["has_ocr_transcription"] else "No",
+            )
+            .add_field(
+                name="Archived", value="Yes" if self.submission["archived"] else "No"
+            )
+        )
+
+        if self.transcription:
+            embed.description = limit_str(self.transcription["text"], 200)
+        if self.submission.get("content_url"):
+            embed.set_image(url=self.submission["content_url"])
+        if self.submission.get("tor_url"):
+            embed.add_field(
+                name="ToR Post", value=f"[Link]({self.submission['tor_url']})"
+            )
+        if self.submission.get("url"):
+            embed.add_field(
+                name="Partner Post", value=f"[Link]({self.submission.get('url')})"
+            )
+        if tr_link:
+            embed.add_field(name="Transcription", value=tr_link)
+        subreddit = (
+            self.submission["url"].split("/")[4] if self.submission["url"] else None
+        )
+        if subreddit:
+            embed.set_author(
+                name=f"r/{subreddit}", url=f"https://reddit.com/r/{subreddit}"
+            )
+
+        return embed
+
+    def _clear(self) -> None:
+        """Reset the local scope caches for a new search."""
+        self.transcription: Union[Dict, None] = None
+        self.submission: Union[Dict, None] = None
+        self.author: Union[Dict, None] = None
+
+    def get_transcription(self) -> None:
+        """Get the target transcription from Blossom."""
+        for option in self.submission["transcription_set"]:
+            response = self.blossom_api.get_transcription(id=get_id_from_url(option))
+            logging.info(response.data[0])
+            if response.data[0]["author"] == self.submission["completed_by"]:
+                self.transcription = response.data[0]
+                return
```

Comment:
Here the behavior is a bit different: Before, when no user completed the post yet, it would return the OCR transcription instead. Additionally this will probably take more API calls, because if it has OCR you might need 2 calls intead of one. If you want to keep it simple, it might be sufficient to change it to a reversed iteration instead, I am guessing that the user transcription will be the last in the array because it was added after the OCR (but I didn't check that yet).

Also, we should probably remove that log statement or change it to `debug` level.

Reply:
Yep, totally forgot about that logging statement. Will fix.

I know that it will take additional calls, but I'm not particularly worried about it because it's one extra call, and my perception is that an extra call won't impact performance enough to worry about it.

If returning the OCR text is desired, that should be easy enough to add -- I'll take a look.

Reply:
Fair, I'm still brand-marked from the Reddit API where an additional API call could mean a whole second of added waiting time

Reply:
Doing this entire lookup process takes between 0.9 and 1 seconds. Given the frequency that I expect this to be used and the actual act of using the bot, I am okay with that delay.

Reply:
I thought it would be more efficient adding a `reversed` to the iteration, but it looks like that new transcriptions (i.e. from an actual user) come first in the response array. So in pretty much all cases this should actually only need one API call, which is nice.